bokehjs/src/lib/models/glyphs/webgl/dash_cache.ts

```diff
@@ -156,9 +157,9 @@ export class DashCache {
   }
 
   public get(pattern: number[]): DashReturn {
-    // Limit pattern to even number of items.
+    // Odd-length patterns are repeated to match canvas.
     if (pattern.length % 2 == 1)
-      pattern = pattern.slice(0, -1)
+      pattern = concat([pattern, pattern])
```

Comment:
I was wondering why this is necessary altogether, but `visuals/line` does only partial normalization and lets canvas to do final normalization.